File: cpp/q1.cpp
//
// Created by Danny Feng on 5/30/24.
//
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashmap;
        for (int i = 0; i < nums.size(); i++) {
            int valueToCheck = target - nums[i];
            if (hashmap.find(valueToCheck) != hashmap.end()) {
                return {hashmap[valueToCheck], i};
            }
            hashmap[nums[i]] = i;
        }
        return {};
    }

};

File: java/src/q1.java
import java.util.HashMap;
import java.util.Map;

class q1 {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numsPos = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            numsPos.put(nums[i], i);
        }
        for (int i = 0; i < nums.length; i++) {
            Integer second = numsPos.get(target - nums[i]);
            if (second != null && second != i ) {
                int[] result = {i, second};
                return result;
            }
        }
        return new int[]{};
    }
}